test/smoke/dashCore.js

```diff
@@ -1,18 +1,79 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+const SSH = require('simple-ssh');
+const fs = require('fs');
+const utils = require('../testUtils');
 
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
-  });
-  describe('Masternodes', () => {
-    // masternode status
-    it('should be masternodes');
-  });
-  describe('Miners', () => {
-    // waitfornewblock
-    it('should mine blocks');
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
```

Comment:
Something wrong with stucture here. "All nodes" should test all DC nodes, not just masternodes: wallet-nodes + full-nodes + miners + masternodes. Miners should test only miners.

Reply:
done. though rpc reuests to miner hosts stuck now. any ideas why it could be?

Reply:
Because of problems with new version of DC. Just ignore it for a while please.